example/layout/Layout.tsx

```diff
@@ -13,23 +12,36 @@ interface LayoutProps {
 
 const Layout = ({ title, children }: LayoutProps): ReactElement => {
   const { locale } = useRouter();
-  const canonicalUrl = useCanonicalUrl(locale);
+  const messages = require(`./layout.${locale}.properties`).default;
```

Comment:
@eemeli exactly, I think it's a good compromise since other solutions I have seen would have a huge file full of strings (possibly for the entire site). At least with this solution, the strings that are loaded are only the ones on the page's loaded components. The idea would be to start with this and then explore [option 3](https://github.com/Avansai/next-intl-routes/pull/5#discussion_r611007272).

Reply:
Good if that's the intent. Probably best to check that your expectations match what actually goes over the wire when you open a page?

Reply:
@eemeli I did a spot check both in the `.next` folder and in Chrome and unless I am missing something, the strings seem to be loaded in the "<page>.js" bundle which only includes the components loaded on that page. This is the behavior I was expecting and in my opinion, I am not sure which one is more performant.. this or having a huge string file for the entire app :)